sqlglot/dialects/hive.py

```diff
@@ -283,6 +283,7 @@ class Generator(generator.Generator):
         JOIN_HINTS = False
         TABLE_HINTS = False
         INDEX_ON = "ON TABLE"
+        IDENTIFIER_CAN_START_WITH_DIGIT = True
```

Comment:
let's move this out to the dialect? because it's in both tokenizer and generator, and then it can get passed in to both

Reply:
yeah, good point

Reply:
done